src/Discovery/DiscoveryActionBar.tsx

```diff
@@ -27,8 +29,117 @@ interface Props {
+    (study) => {
+      study[manifestFieldName].forEach(
+        (file) => {
+          downloadSize += file.file_size;
+        },
+      );
+    },
+  );
+
+  if (downloadSize > maxDownloadSizeBytes) {
+    const maxSizeMB = (maxDownloadSizeBytes / 1000000).toFixed(1);
+    const downloadSizeMb = (downloadSize / 1000000).toFixed(1);
+    setDownloadStatusMessage(
+      {
+        title: 'Download limit exceeded',
+        message: `
+        The selected studies contain ${downloadSizeMb} MB of data, which exceeds the download limit of ${maxSizeMB} MB.
+        Please deselect some studies and try again, or use the gen3 client.
+        `,
+        active: true,
+      },
+    );
+    return;
+  }
+
+  setDownloadInProgress(true);
+  setDownloadStatusMessage(
+    {
+      title: 'Your download is being prepared',
+      message: 'Please remain on this page while your download is being prepared.\n\n'
+                 + 'When your download is ready, it will begin automatically. You can close this window.',
+      active: true,
+    },
+  );
+
+  const triggerDownloadResponse = await fetch(
+    '/job/dispatch',
+    {
+      method: 'POST',
+      body: JSON.stringify({
+        action: 'batch-export',
+        input: {
+          study_ids: studyIDs,
+        },
+      }),
+    },
+  );
+  const initialialJobState = await triggerDownloadResponse.json();
+  const downloadFailedMessage = {
+    title: 'Download failed',
+    message: 'There was a problem preparing your download.'
+               + 'Please consider using the gen3 client to download these files via a manifest.',
+    active: true,
+  };
+
+  if (initialialJobState === null) {
+    console.warn('Sower is not configured for batch-export job.');
+    setDownloadInProgress(false);
+    setDownloadStatusMessage(downloadFailedMessage);
+  } else {
+    const { uid } = initialialJobState;
+    const pollForUpdate = async () => {
+      const statusResponse = await fetch(`/job/status?UID=${uid}`);
+      const statusObject = await statusResponse.json();
+      const { status } = statusObject;
+
+      if (status === 'Failed') {
+        setDownloadStatusMessage(downloadFailedMessage);
+        setDownloadInProgress(false);
+      } else if (status === 'Completed') {
+        const outputResponse = await fetch(`/job/output?UID=${uid}`);
+        const outputJSON = await outputResponse.json();
+        const url = outputJSON.output;
+        const message = `Your download has been prepared. If your download doesn't start automatically, please copy and paste this url into your browser:\n\n${url}`;
+        setDownloadStatusMessage(
+          {
+            title: 'Your download is ready',
+            message,
+            active: true,
+          },
+        );
+        setDownloadInProgress(false);
+        setTimeout(() => window.open(url), 2000);
+      } else {
+        setTimeout(pollForUpdate, 5000);
+      }
+    };
+    setTimeout(pollForUpdate, 5000);
```

Comment:
use `setTimeout()` for the final `window.open()` makes sense, but for `pollForUpdate`, why don't use `setInterval()` and use the interval as a React state?